File: Codechef/AprilChallenge2012/PLAYFIT.cc
#include <cstdio>
#include <algorithm>

using namespace std;

int main() {
    int T;
    scanf("%d", &T);

    while(T--) {
        int N;
        scanf("%d", &N);

        int ans = 0;
        int minval = 0;
        for(int i = 0; i < N; ++i) {
            int v;
            scanf("%d", &v);
            if(i == 0) {
                minval = v;
            }
            else {
                minval = min(minval, v);
                int diff = v-minval;
                ans = max(ans, diff);
            }
        }

        if(ans == 0) {
            puts("UNFIT");
        }
        else {
            printf("%d\n", ans);
        }
    }

    return 0;
}

File: Codechef/AprilChallenge2012/PLAYFIT.rb
#!/usr/bin/ruby

gets.to_i.times do
  n = gets.to_i
  arr = gets.chomp.split.map(&:to_i)

  ans = 0
  min = nil
  arr.each do |g|
    if min.nil?
      min = g
    else
      min = g if g < min
      diff = g - min
      ans = diff if diff > ans
    end
  end

  if ans == 0
    ans = "UNFIT"
  end

  puts ans
end
